scripts/popup.js

```diff
@@ -1,364 +1,365 @@
+              })
             }
-          });
+          })
         }
       }
-    });
-  });
+    })
+  })
 }
 
-function show_news(){
-    chrome.tabs.query({active: true,currentWindow:true},function(tabs){
-        url=tabs[0].url;
-        var to_check_url=url.replace(/^https?:\/\/(www\.)?/,'');
-        var news_host=to_check_url.split(".")[0];
-        tabId=tabs[0].id;
-        var set_of_sites = new Set([
-          "apnews",
-          "factcheck",
-          "forbes",
-          "huffingtonpost",
-          "nytimes",
-          "politifact",
-          "snopes",
-          "theverge",
-          "usatoday",
-          "vox",
-          "washingtonpost",
-        ]);
-        chrome.storage.sync.get(['news'],function(event){
-            if(event.news==true){
-                if(set_of_sites.has(news_host)){
-                    document.getElementById('news_recommend_tr').style.display="block";
-                    document.getElementById('news_recommend_tr').onclick=function(){
-                        chrome.storage.sync.get(['show_context'],function(event1){
-                            if(event1.show_context==undefined){
-                                event1.show_context="tab";
-                            }
-                            if(event1.show_context=="tab"){
-                                chrome.tabs.create({url:chrome.runtime.getURL("recommendations.html")+"?url="+url});
-                            }else{
-                              chrome.system.display.getInfo(function(displayInfo){
-                                let height = displayInfo[0].bounds.height;
-                                let width = displayInfo[0].bounds.width;
-                                chrome.windows.create({url:chrome.runtime.getURL("recommendations.html")+"?url="+url,width:width/2, height:height, top:0, left:width/2, focused:true});
-                              });
-                            }
-                        });
-                    }
-                }
-            }
-        });
-    });
+function show_news () {
+  chrome.tabs.query({ active: true, currentWindow: true }, function (tabs) {
+    url = tabs[0].url
+    var to_check_url = url.replace(/^https?:\/\/(www\.)?/, '')
+    var news_host = to_check_url.split('.')[0]
+    tabId = tabs[0].id
+    var set_of_sites = new Set([
+      'apnews',
+      'factcheck',
+      'forbes',
+      'huffingtonpost',
+      'nytimes',
+      'politifact',
+      'snopes',
+      'theverge',
+      'usatoday',
+      'vox',
+      'washingtonpost'
+    ])
+    chrome.storage.sync.get(['news'], function (event) {
+      if (event.news) {
+        if (set_of_sites.has(news_host)) {
+          $('#news_recommend_tr').show().click(() => {
+            chrome.storage.sync.get(['show_context'], function (event1) {
+              if (event1.show_context === undefined) {
+                event1.show_context = 'tab'
+              }
+              if (event1.show_context === 'tab') {
```

Comment:
You can do
```
if (event1.show_context === 'tab' || event1.show_context === undefined)
```
and drop the previous 3 lines of code.

Reply:
This shows up often throughout the entire codebase.  I'll keep my eye out for it. Or maybe make a function in utils and just call that.